packages/s3-replicator/iam.tf

```diff
@@ -0,0 +1,51 @@
+data "aws_iam_policy_document" "assume_lambda_role" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions   = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "replicator_lambda_role" {
+  name                 = "${var.prefix}_replicator_lambda_role"
+  assume_role_policy   = data.aws_iam_policy_document.assume_lambda_role.json
+  permissions_boundary = var.permissions_boundary
+}
+
+data "aws_iam_policy_document" "replicator_policy_document" {
+  statement {
```

Comment:
Do we need all of these permissions?  I would guess ec2 and logs are not needed?

Reply:
EC2 network interface permissions are necessary to run the lambda in a VPC. Logs perms are needed to write cloudwatch logs for error logging. The latter is optional I guess, but in that case there will be no logging whatsoever.